### DISCUSSION TITLE
Torsional Vibration

### DISCUSSION THREAD

**Discussion by rahultarale** [petrobras/ross#949] [Q&A]
Can we perform torsional vibration analysis of two shafts using coupling/gear stiffness values? If YES, please can you share one complete example of it including plotting Campbell diagram, mode shapes and steady state torque conditions?

**Comment by raphaeltimbo** [comment]
Hi @rahultarale!
There is some good discussion about torsional dynamics in [this question](https://github.com/petrobras/ross/discussions/795).

At the moment, ROSS supports torsional analysis using ShaftElement6DoF and BearingElement6DoF elements. However, we have some limitations on the results we can extract from the model. For instance, the Campbell diagram is only available for the 4 degrees of freedom model, which means we cannot generate this diagram with lateral and torsional vibration modes.

Regarding coupling/gear stiffness values, you would need to create two rotors and an additional matrix to connect them with the appropriate stiffness values.

Although some colleagues have already performed this analysis, the code was not integrated into the ROSS library. I will investigate if we can incorporate these functionalities in the library soon.

@atbrandao

**Reply by rahultarale** [reply]
Hi @raphaeltimbo,

Thanks for the information. I will be waiting for the library integration ASAP.

**Reply by evariste-b** [reply]
Hi, @raphaeltimbo 

I'm also interested in the torsional analysis of two shaft connected by a gearbox.
Do you have any update about this topic since it was previously discussed here ?

Otherwise, could you please give me more details for the method you suggested with the addition of a matrix to connect both shafts ?
How do you define the matrix in ross, and how do you use it in the shaft modeling to connect the 2 shafts ?

Thanks if you can help me with this problem !

**Reply by shishirsanjeeva** [reply]
Hi @raphaeltimbo 
I am also interested in the torsional analysis of two shaft connected by a gearbox. Do you have any update about this topic?

**Reply by jguarato** [reply]
Hi @evariste-b and @shishirsanjeeva

We are currently developing (#1111) a new class, called `MultiRotor`, for connecting shafts by gearbox. We hope to release the changes soon after testing is complete.
